clients/seating/src/App.tsx

```diff
@@ -1,45 +1,126 @@
-import React from 'react';
+import * as React from 'react';
 import {
-  Button,
   withStyles,
-  Theme,
   createStyles,
   WithStyles,
+  Grid,
+  Hidden,
 } from '@material-ui/core';
 
-const styles = (theme: Theme) =>
+import Layout from './components/Layout';
+import Navbar from './components/Navbar';
+import { Section, fetchSeatingLayout } from './mocks';
+import Checkbox from '@material-ui/core/Checkbox';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
+import ListItemText from '@material-ui/core/ListItemText';
+import Divider from '@material-ui/core/Divider';
+import Reserve from './Dialogs/Reserve';
+import ListCreate from './components/List';
+import Button from '@material-ui/core/Button';
+
+const styles = () =>
   createStyles({
-    root: {
-      textAlign: 'center',
-      marginTop: '400px',
+    main: {
+      display: 'flex',
+      height: '100vh',
     },
-    apple: {
-      margin: theme.spacing.unit * 3,
+    content: {
+      flexGrow: 1,
     },
   });
 
-type Props = WithStyles<typeof styles>;
+type AppProps = WithStyles<typeof styles>;
+
+const App: React.FC<AppProps> = ({ classes }) => {
+  const [seatingLayoutYOffset, setSeatingLayoutYOffset] = React.useState(0);
+  const [sections, setSections] = React.useState<Section[]>([]);
+  const [isLoading, setLoading] = React.useState(true);
+  const [selectedTable, setSelectedTable] = React.useState<number | null>(null);
 
-const App: React.FC<Props> = ({ classes }) => {
-  const [count, setCount] = React.useState(0);
+  const navbarRef = React.useRef<HTMLDivElement | null>(null);
 
-  const handleClick = () => {
-    setCount((prevCount) => (prevCount + 1) % 10);
+  const handleTableClick = (tableId: number) => () => {
+    setSelectedTable(tableId);
   };
 
+  React.useLayoutEffect(() => {
+    if (navbarRef.current) {
+      const navBarHeight = navbarRef.current.getBoundingClientRect().height;
+      setSeatingLayoutYOffset(navBarHeight);
+    }
+  });
+
+  React.useEffect(() => {
+    fetchSeatingLayout().then((data) => {
```

Comment:
I would prefer async await. Also, it might be smart to add error handling for the future.

Reply:
We'll probably extract this to a custom hook and have all the fetching and error logic encapsulated in that, but `useEffect` can't return a promise so that's why I'm using `.then`